lib/matplotlib/tests/test_axes.py

```diff
@@ -3981,6 +3981,11 @@ def test_errorbar_nan(fig_test, fig_ref):
     ax.errorbar([4], [3], [6], fmt="C0")
 
 
+def test_errorbar_allnan_error():
+    fig, ax = plt.subplots()
+    ax.errorbar([0], [0], [np.nan])
```

Comment:
Thank you for this!

```suggestion
    with pytest.warns(RuntimeWarning, match="All-NaN axis encountered"):
        ax.errorbar([0], [0], [np.nan])
```
I think this is the way to go about it now. I cannot really see the consequences later on, but at least it will now give a warning rather than the not so clear StopIteration.

I also suggest checking if this works for #18294 as well and in that case add a test to close it.